pyproject.toml

```diff
@@ -32,13 +32,15 @@ exclude = [
 # are invoked via separate runs of ruff in pre-commit:
 # see our .pre-commit-config.yaml file for details
 exclude = ["**/test_cases/**/*.py"]
-select = [
+external = ["F821", "NQA", "Y"]
+extend-select = [
```

Comment:
What effect does switching from `select` to `extend-select` have here?

Reply:
> What effect does switching from `select` to `extend-select` have here?

https://docs.astral.sh/ruff/rules/
> By default, Ruff enables Flake8's F rules, along with a subset of the E rules, omitting any stylistic rules that overlap with the use of a formatter, like ruff format or [Black](https://github.com/psf/black).

https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules
> None of the above are included in Ruff's default configuration. However, if you've enabled any of these rules or their parent categories (like Q), we recommend disabling them via the linter's [lint.ignore](https://docs.astral.sh/ruff/settings/#lint_ignore) setting.

Reply:
That makes sense. Is `extend-select` still completely overridden if you specify `--select` on the command line, like we do in our pre-commit config?

Reply:
I wanted to say yes, but reading https://docs.astral.sh/ruff/linter/#rule-selection makes it sounds like `extend-select` is always applied after even a "highest priority `select`" (which is the CLI), then the `ignore`.

> To resolve the enabled rule set, Ruff may need to reconcile [lint.select](https://docs.astral.sh/ruff/settings/#lint_select) and [lint.ignore](https://docs.astral.sh/ruff/settings/#lint_ignore) from a variety of sources, including the current pyproject.toml, any inherited pyproject.toml files, and the CLI (e.g., [--select](https://docs.astral.sh/ruff/settings/#lint_select)).

> 
> In those scenarios, Ruff uses the "highest-priority" [select](https://docs.astral.sh/ruff/settings/#lint_select) as the basis for the rule set, and then applies [extend-select](https://docs.astral.sh/ruff/settings/#lint_extend-select) and [ignore](https://docs.astral.sh/ruff/settings/#lint_ignore) adjustments. CLI options are given higher priority than pyproject.toml options, and the current pyproject.toml file is given higher priority than any inherited pyproject.toml files.

There's also less formatter-conflicting rules than I remembered (at least when it comes to `F` and `E`. So I might just do that instead.